src/backends/policy_engine/souffle/check_policy_compliance.cc

```diff
@@ -27,11 +27,14 @@
 #include "src/backends/policy_engine/policy.h"
 #include "src/backends/policy_engine/souffle/souffle_policy_checker.h"
 #include "src/common/logging/logging.h"
+#include "src/ir/proto_to_ir.h"
 #include "src/parser/ir/ir_parser.h"
 
-ABSL_FLAG(std::string, ir, "", "the IR file");
+// TODO(cromwellian): why doesn't std::optional<std::string> (or absl::optional) work?
+ABSL_FLAG(std::optional<std::string>, ir, std::nullopt, "the IR file");
 ABSL_FLAG(std::string, sql_policy_rules, "",
           "file containing the SQL policy rules");
+ABSL_FLAG(std::optional<std::string>, proto, std::nullopt, "the proto file");
```

Comment:
I think this would be better as a command (i.e. always take 1 file and then 0 or 1 commands either 'proto' or 'ir').

If I understand correctrly, currently your flags will accept a null `FLAG_ir` and a null `FLAG_proto` and then segfault as the `ir_*` unique_ptrs are not initialized.

Reply:
I'll fix the segfault, but I think we should discuss how to arrange the CLI. I kind of feel that the ideal thing would be just to detect the file type (proto or IR format) and work no matter what.  Either that, or build two different CLI binaries, one that handles each format. For now, just leave as two flags (with fixes to remove foot guns)